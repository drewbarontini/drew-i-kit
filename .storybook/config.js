import { configure, addDecorator } from '@storybook/react';
import React, { Fragment } from 'react';
import { ThemeProvider } from 'styled-components';
import 'normalize.css';

import { theme } from '../config';

import GlobalStyles from '../components/GlobalStyles';

const req = require.context('../components/', true, /.story.js$/);
function loadStories() {
  req.keys().forEach(filename => req(filename));
}

function withStyledComponents(story) {
  return (
    <ThemeProvider theme={theme}>
      <Fragment>
        {story()}
        <GlobalStyles />
      </Fragment>
    </ThemeProvider>
  );
}

addDecorator(withStyledComponents);
configure(loadStories, module);
