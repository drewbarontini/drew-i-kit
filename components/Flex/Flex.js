import styled from 'styled-components';

import { setProp } from '../../lib/styleUtils';

const Flex = styled.div`
  display: flex;

  ${setProp({ prop: 'alignItems', cssProp: 'align-items' })};
  ${setProp({ prop: 'flex', cssProp: 'flex' })};
  ${setProp({ prop: 'flexDirection', cssProp: 'flex-direction' })};
  ${setProp({ prop: 'justifyContent', cssProp: 'justify-content' })};
  ${setProp({ prop: 'flexWrap', cssProp: 'flex-wrap' })};
`;

Flex.displayName = 'Flex';

export default Flex;
