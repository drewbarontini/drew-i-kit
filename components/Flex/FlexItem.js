import PropTypes from 'prop-types';
import styled from 'styled-components';

import { setProp } from '../../lib/utils';

const FlexItem = styled.div`
  ${setProp({
    prop: 'size',
    cssProp: 'flex-basis',
  })};
  ${setProp({
    prop: 'margin',
    themeKey: 'spacing',
  })};
  ${setProp({
    prop: 'marginRight',
    themeKey: 'spacing',
  })};
`;

FlexItem.propTypes = {
  margin: PropTypes.string,
  marginRight: PropTypes.string,
  size: PropTypes.string,
};

FlexItem.displayName = 'FlexItem';

export default FlexItem;
