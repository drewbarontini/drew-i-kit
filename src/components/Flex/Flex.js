import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';
import { ifProp } from 'styled-tools';

import { setProp } from '../../lib/props';
import { responsive } from '../../lib/responsive';

const Flex = styled.div`
  display: flex;

  ${ifProp(
    'center',
    css`
      align-items: center;
      justify-content: center;
    `
  )};

  ${setProp({ prop: 'alignItems' })};
  ${setProp({ prop: 'flex' })};
  ${setProp({ prop: 'flexDirection' })};
  ${setProp({ prop: 'flexWrap' })};
  ${setProp({ prop: 'height' })};
  ${setProp({ prop: 'justifyContent' })};

  ${responsive};
`;

Flex.displayName = 'Flex';

Flex.propTypes = {
  alignItems: PropTypes.string,
  center: PropTypes.bool,
  flex: PropTypes.string,
  flexDirection: PropTypes.string,
  flexWrap: PropTypes.string,
  height: PropTypes.string,
  justifyContent: PropTypes.string,
};

export default Flex;
