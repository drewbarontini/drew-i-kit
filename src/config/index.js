export { default as cssProps } from './cssProps';
export { default as icons } from './icons';
export { default as theme } from './theme';
