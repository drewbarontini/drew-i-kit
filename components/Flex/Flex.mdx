---
name: Flex
route: /components/flex
---

# Flex

import Box from '../Box';
import Flex, { FlexItem } from './';

<Flex>
  <FlexItem marginRight="base">
    <Box height="200px" width="200px" backgroundColor="primary" />
  </FlexItem>
  <FlexItem marginRight="base">
    <Box height="200px" width="200px" backgroundColor="primary" />
  </FlexItem>
  <FlexItem>
    <Box height="200px" width="200px" backgroundColor="primary" />
  </FlexItem>
</Flex>
