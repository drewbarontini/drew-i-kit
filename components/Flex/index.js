import withThemeProvider from '../withThemeProvider';
import Flex from './Flex';
import FlexItem from './FlexItem';

const DecoratedFlex = withThemeProvider(Flex);

export { DecoratedFlex as default, FlexItem };
