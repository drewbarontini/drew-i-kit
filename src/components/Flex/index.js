import Flex from './Flex';
import FlexItem from './FlexItem';

export { Flex as default, FlexItem };
