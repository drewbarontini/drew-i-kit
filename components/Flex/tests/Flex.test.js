import React from 'react';
import { render } from 'react-testing-library';
import 'jest-styled-components';

import { theme } from '../../../config';

import Flex, { FlexItem } from '../';

describe('<Flex />', () => {
  it('should render properly', () => {
    const { container } = render(
      <Flex theme={theme}>
        <FlexItem>Hello</FlexItem>
        <FlexItem>Flex</FlexItem>
      </Flex>
    );
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toHaveStyleRule('display', 'flex');
  });
});
