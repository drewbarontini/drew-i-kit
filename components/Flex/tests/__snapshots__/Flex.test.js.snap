// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Flex /> should render properly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
  class="c0"
>
  <div
    class=""
  >
    Hello
  </div>
  <div
    class=""
  >
    Flex
  </div>
</div>
`;
